src/puppet-padplus.ts

```diff
@@ -380,14 +380,18 @@ export class PuppetPadplus extends Puppet {
       if (searchContact === null) {
         return null
       }
-      const contactPayload = convertSearchContactToContact(searchContact, isPhoneNumber)
-
-      if (this.manager && this.manager.cacheManager) {
-        await this.manager.cacheManager.setContact(phone, contactPayload)
-        return phone
-      } else {
+      if (!this.manager || !this.manager.cacheManager) {
         throw new Error(`no cache manager`)
       }
+
+      const cacheContactPayload = await this.manager.cacheManager.getContact(searchContact.v1)
```

Comment:
If the cache been removed, some mistakes will happen.

Maybe we could use `isStrangerV1()` to justify the `searchContact.v1` is `v1` or `wxid`, we can return `searchContact.v1` if the result of `isStrangerV1()` is true .

The test result show:
friend: v1 -> wxid
not friend: v1 -> v1_xxxxxxxx